lib/srv/desktop/windows_server.go

```diff
@@ -202,16 +202,10 @@ type WindowsServiceConfig struct {
 	// If empty LDAP address will be used.
 	// Used for NLA support when AD is true.
 	KDCAddr string
-	// DiscoveryBaseDN is the base DN for searching for Windows Desktops.
-	// Desktop discovery is disabled if this field is empty.
-	DiscoveryBaseDN string
-	// DiscoveryLDAPFilters are additional LDAP filters for searching for
-	// Windows Desktops. If multiple filters are specified, they are ANDed
-	// together into a single search.
-	DiscoveryLDAPFilters []string
-	// DiscoveryLDAPAttributeLabels are optional LDAP attributes to convert
-	// into Teleport labels.
-	DiscoveryLDAPAttributeLabels []string
+	// Discovery contains policies for configuring LDAP-based discovery.
+	Discovery []servicecfg.LDAPDiscoveryConfig
+	// DiscoveryInterval configures how frequently the discovery process runs.
+	DiscoveryInterval time.Duration
```

Comment:
Do we need to check and set default or will this always be defaulted by `applyWindowsDesktopConfig`?

Reply:
Always defaulted by `applyWindowsDesktopConfig`, but that felt a bit off so I changed the approach based on your comment.

Now, `applyWindowsDesktopConfig` will disallow a negative interval, but it will be okay with 0 (empty), and then the standard check and set default will handle the empty case.